src/plugins/intel_gna/preprocessing.hpp

```diff
@@ -14,12 +14,22 @@ void ConvertToInt16(int16_t *ptr_dst,
                     const uint32_t num_columns,
                     const float scale_factor);
 
-void ConvertToFloat(float *ptr_dst,
-                    int32_t *ptr_src,
-                    const uint32_t num_rows,
-                    const uint32_t num_columns,
-                    const float scale_factor);
-
 int16_t ConvertFloatToInt16(float src);
 int8_t ConvertFloatToInt8(float src);
+
+template<typename T1, typename T2>
+inline void CastPrecision(T2 *ptr_dst, T1 *ptr_src, const uint32_t num_rows, const uint32_t num_columns,
```

Comment:
I would name this function 
"Unscale/Rescale And Cast" rather than "Cast" itself

Reply:
yes and no, we are also casting the output type + unscale it, I can rename it to UnscaleAndCast()?

Reply:
Updated to UnscaleAndCast

Reply:
Great